### PR TITLE
fix(masking): mask email while logging SQL query

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Bugfix

## Description

Pass the masked email in the SQL query function.

## Motivation and Context

This fixes plaintext logging of email during find and update users by email. In this change instead of passing the plain-text email into the function, changed it to masked email which will mask the email while logging.

## How did you test it?

- Signup a new user
```bash
curl --location 'http://localhost:8080/user/signup' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "mani@test.in",
    "password": "password"
}'
```

 - Signin the user
```bash
curl --location 'http://localhost:8080/user/v2/signin' \
--header 'Content-Type: application/json' \
--header 'Cookie: login_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiOTg4N2YzNzQtNDYyZC00MDU1LWE4NGEtZDY5MzU3NTNmZjY2IiwibWVyY2hhbnRfaWQiOiJtZXJjaGFudF8xNzEzODYyNzU2Iiwicm9sZV9pZCI6Im9yZ19hZG1pbiIsImV4cCI6MTcxNDAzNTU4MCwib3JnX2lkIjoib3JnX3hjVDhqaUdZWTAza2pqUFlHakt4In0._xY56Sv8A3doS6NY00eSqRJxoQA3i-koNkcvtbdv4fI' \
--data-raw '{
    "email": "mani@test.in",
    "password": "password"
}'
```
- Search the logs if plain text email is logged or not. In grafana, data source as loki
```
{app="bach"} |~ `^[\w\-\.]+@([\w-]+\.)+[\w-]{2,}$`
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

**Review by lsampras — DISMISSED** [review]

**Review by ThisIsMani — DISMISSED** [review]

File: crates/router/src/core/user.rs [hunk 1]
```diff
@@ -389,7 +386,9 @@ pub async fn reset_password(
     let user = state
         .store
         .update_user_by_email(
-            email_token.get_email(),
+            &email_token
+                .get_email()
+                .change_context(UserErrors::EmailParsingError)?,
```
Comment (racnan) [inline crates/router/src/core/user.rs:391]: this should be `500`.

File: crates/router/src/core/user.rs [hunk 2]
```diff
@@ -949,7 +948,11 @@ pub async fn accept_invite_from_email(
 
     let user: domain::UserFromStorage = state
         .store
-        .find_user_by_email(email_token.get_email())
+        .find_user_by_email(
+            &email_token
+                .get_email()
+                .change_context(UserErrors::EmailParsingError)?,
```
Comment (racnan) [inline crates/router/src/core/user.rs:954]: same `500`

File: crates/router/src/core/user.rs [hunk 3]
```diff
@@ -1326,7 +1329,11 @@ pub async fn verify_email_without_invite_checks(
     auth::blacklist::check_email_token_in_blacklist(&state, &token).await?;
     let user = state
         .store
-        .find_user_by_email(email_token.get_email())
+        .find_user_by_email(
+            &email_token
+                .get_email()
+                .change_context(UserErrors::EmailParsingError)?,
```
Comment (racnan) [inline crates/router/src/core/user.rs:1335]: `500`

File: crates/router/src/core/user.rs [hunk 4]
```diff
@@ -1362,7 +1369,11 @@ pub async fn verify_email(
 
     let user = state
         .store
-        .find_user_by_email(email_token.get_email())
+        .find_user_by_email(
+            &email_token
+                .get_email()
+                .change_context(UserErrors::EmailParsingError)?,
```
Comment (racnan) [inline crates/router/src/core/user.rs:1375]: `500`

**Review by racnan — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Pull request merged**